lenskit/algorithms/basic.py

```diff
@@ -538,3 +292,18 @@ def recommend(self, user, n=None, candidates=None, ratings=None):
 
     def __str__(self):
         return 'Random'
+
+
+class KnownRating(Predictor):
+    """
+    The known rating algorithm memorizes ratings provided in the fit method.
+    """
+
+    def fit(self, ratings, **kwargs):
+        self.ratings = ratings
```

Comment:
We should consider setting up indexes so this in a consistent state, and then using them in `predict_for_user`.

Reply:
A `MultiIndex` on `[user, item]` will do the trick, then sort by index (`sort_index`). This class is simple enough we won't have the problems mutli-indexes usually have.